src/plugins/pcr_primer_design_for_dna_assembly/src/utils/UnwantedConnectionsUtils.cpp

```diff
@@ -29,62 +29,83 @@
-                                                     double unwantedMeltingTemperature,
-                                                     int unwantedDimerLength,
-                                                     QString &report) {
+                                                     const ExclusionCriteria& criterion,
+                                                     const OptionalParametersToExclude& optionalThresholds,
+                                                     QString& report) {
     PrimersPairStatistics calc(forwardSequence, reverseSequence);
     report = PCRPrimerDesignForDNAAssemblyPlugin::tr("<b>Hetero-dimer:</b><br>");
-    return areUnwantedParametersPresentedInDimersInfo(calc.getDimersInfo(), unwantedDeltaG, unwantedMeltingTemperature,
-                                                      unwantedDimerLength, report);
+    return areUnwantedParametersPresentedInDimersInfo(calc.getDimersInfo(), criterion, optionalThresholds, report);
 }
 
 bool UnwantedConnectionsUtils::areUnwantedParametersPresentedInDimersInfo(const DimerFinderResult& dimersInfo,
-                                                                          double unwantedDeltaG,
-                                                                          double unwantedMeltingTemperature,
-                                                                          int unwantedDimerLength,
-                                                                          QString &report) {
-    if (dimersInfo.dimersOverlap.isEmpty()) {
+                                                                          const ExclusionCriteria& criterion,
+                                                                          const OptionalParametersToExclude& thresholds,
+                                                                          QString& report) {
+    if (dimersInfo.dimersOverlap.isEmpty() || !thresholds.isAnyParameterSet()) {
         return false;
     }
-    double dimerMeltingTemp = PrimerStatistics::getMeltingTemperature(dimersInfo.dimer.toLocal8Bit());
-    int dimerLength = dimersInfo.dimer.length();
-    bool isDeltaGUnwanted = dimersInfo.deltaG <= unwantedDeltaG;
-    bool isMeltingTemperatureUnwanted = unwantedMeltingTemperature <= dimerMeltingTemp;
-    bool isLengthUnwanted = unwantedDimerLength <= dimerLength;
-    bool isUnwantedParameter = isDeltaGUnwanted && isMeltingTemperatureUnwanted && isLengthUnwanted;
+
+    QList<bool> conditionResult;
+    if (thresholds.gibbsFreeEnergy.isValid()) {
+        conditionResult += dimersInfo.deltaG <= thresholds.gibbsFreeEnergy.toDouble();
+    }
+    if (thresholds.meltingPoint.isValid()) {
+        QByteArray dimer = dimersInfo.dimer.toLocal8Bit();
+        double dimerMeltingTemp = PrimerStatistics::getMeltingTemperature(dimer);
+        double reverseDimerMeltingTemp =
+            PrimerStatistics::getMeltingTemperature(DNASequenceUtils::reverseComplement(dimer));
```

Comment:
The temperature of the reverse complementary dimer is not added to the report and the log. Add?